homework-completed-w3

mahmoudbd opened the pull request.

mogaleaf commented on Week3/Normalization/Normalization.md:
```diff
@@ -0,0 +1,142 @@
+Exercise 1 : Normalization   
```
Your tables are good in the pdf except the "fourth table" (table should not be names first,second,third, fourth of course :) )
You need more than four table, because dinner contains foods and dinner contains members, so you have more than one relation going on here.

mogaleaf replied:
In fact, I'm coming back on what I said. Your solution works as each member could choose different food in one dinner!

mogaleaf commented on Week3/Transactions/transaction.js:
```diff
@@ -0,0 +1,44 @@
+	const commit = `commit`;
+	connection.connect();
+
+	try {
```
good

mogaleaf commented on Week3/injection/injection.js:
```diff
@@ -0,0 +1,27 @@
+
+function getPopulation(Country, name, code, cb) {
+	connection.query(
+		`SELECT Population FROM ${Country} WHERE Name = +
```
I invite you to read:
https://github.com/mysqljs/mysql#preparing-queries
It's a better syntax.
Plus, you need to escape Country too.
Where is the code of the injection query ?

mahmoudbd commented:
@mogaleaf  first, i would like to thank you about the help  , i added new pdf for Normalization and updated injection exercise, could you please check it ?

mogaleaf commented on Week3/injection/injection.js:
```diff
@@ -1,27 +1,13 @@
-			cb(null, result[0].name);
-		}
-	);
+	connection.query(`SELECT Population FROM ?? WHERE Name = ?AND code = ?`, [ Country, name, code ], function(
```
perfect
